src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
+    } else {
+        globalCtx->nextEntranceIndex = 0xA810;
+    }
+    gSaveContext.nextCutsceneIndex = 0;
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 3;
+    gSaveContext.nextTransition = 3;
+    this->actionFunc = func_80953F8C;
+    gSaveContext.weekEventReg[90] &= 0xBF;
+    func_800FE498();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F14.s")
+void func_80954340(BgIngate* this, GlobalContext* globalCtx) {
+    if (!DECR(this->unk16A)) {
+        if (this->unk164 != 0) {
+            func_800B7298(globalCtx, &this->dyna.actor, 6U);
+            this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+            func_80953F14(this, globalCtx);
+            func_800FE484();
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F8C.s")
+void func_809543D4(BgIngate* this, GlobalContext* globalCtx) {
+    u8 talkState;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F9C.s")
+    talkState = Message_GetState(&globalCtx->msgCtx);
+    if (((talkState == 4) || (talkState == 5)) && func_80147624(globalCtx)) {
+        switch (this->dyna.actor.textId) {
+            case 0x9E4:
+                this->dyna.actor.textId = 0x9E5;
+                func_80151938(globalCtx, this->dyna.actor.textId);
+                break;
+            case 0x9E5:
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_800B7298(globalCtx, &this->dyna.actor, 6);
+                    this->unk160 &= ~0x4;
+                    this->actionFunc = func_809541B8;
+                    func_800FE498();
+                    func_8019F208();
+                } else {
+                    if (this->unk164 != 0) {
+                        this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+                    }
+                    func_80953F14(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
+                    func_8019F230();
+                }
+                func_801477B4(globalCtx);
+                break;
+            case 0x9E6:
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_80953EA4(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
+                    func_8019F208();
+                } else {
+                    this = this;
+                    func_800B7298(globalCtx, &this->dyna.actor, 6);
+                    this->unk160 &= ~0x4;
+                    this->actionFunc = func_809541B8;
+                    func_800FE498();
+                    func_8019F230();
+                }
+                func_801477B4(globalCtx);
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_809541B8.s")
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    BgIngate* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
```

Comment:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
    BgIngate* this = THIS;
```
Does this still match?